brilirs/src/interp.rs

```diff
@@ -516,6 +518,32 @@ fn execute_value_op<T: std::io::Write>(
       let res = Value::Pointer(arg0.add(arg1));
       state.env.set(dest, res);
     }
+    Float2Bits => {
+      let input = get_arg::<&Value>(&state.env, 0, args);
+
+      let (Value::Float(float), Type::Int) = (input, op_type) else {
+        panic!("Invalid bitcast");
+      };
```

Comment:
Maybe this is a hold-over from your previous design but I don't think the explicit panic here is necessary anymore. The op_type is no longer needed at runtime after type-checking here. `get_arg` can get you the float directly